packages/teams-js/test/public/navigation.spec.ts

```diff
@@ -21,134 +22,173 @@ describe('MicrosoftTeams-Navigation', () => {
-      await utils.initializeWithContext('content');
-
-      navigateCrossDomain('https://valid.origin.com');
-    });
+          const returnFocusMessage = utils.findMessageByFunc('returnFocus');
+          expect(returnFocusMessage).not.toBeNull();
+          expect(returnFocusMessage.args.length).toBe(1);
+          expect(returnFocusMessage.args[0]).toBe(true);
+        });
 
-    it('should allow calls from sidePanel context', async () => {
-      await utils.initializeWithContext('sidePanel');
+        it(`navigation.returnFocus should successfully returnFocus when set to false and initialized with ${context} context`, async () => {
+          await utils.initializeWithContext(context);
 
-      navigateCrossDomain('https://valid.origin.com');
-    });
+          returnFocus(false);
 
-    it('should allow calls from settings context', async () => {
-      await utils.initializeWithContext('settings');
+          const returnFocusMessage = utils.findMessageByFunc('returnFocus');
+          expect(returnFocusMessage).not.toBeNull();
+          expect(returnFocusMessage.args.length).toBe(1);
+          expect(returnFocusMessage.args[0]).toBe(false);
+        });
+      } else {
+        it(`navigation.returnFocus should not allow calls when set to true and initialized with ${context} context`, async () => {
+          await utils.initializeWithContext(context);
+          expect(() => returnFocus(true)).toThrow(
+            `This call is only allowed in following contexts: ${JSON.stringify(
+              allowedContexts,
+            )}. Current context: "${context}".`,
+          );
+        });
 
-      navigateCrossDomain('https://valid.origin.com');
+        it(`navigation.returnFocus should not allow calls when set to false and initialized with ${context} context`, async () => {
+          await utils.initializeWithContext(context);
+          expect(() => returnFocus(false)).toThrow(
+            `This call is only allowed in following contexts: ${JSON.stringify(
+              allowedContexts,
+            )}. Current context: "${context}".`,
+          );
+        });
+      }
     });
+  });
 
-    it('should allow calls from remove context', async () => {
-      await utils.initializeWithContext('remove');
-
-      navigateCrossDomain('https://valid.origin.com');
+  describe('Testing navigation.navigateToTab function', () => {
+    it('navigation.navigateToTab should not allow calls before initialization', () => {
+      expect(() => navigateToTab(null)).toThrowError('The library has not yet been initialized');
     });
 
-    it('should allow calls from task context', async () => {
-      await utils.initializeWithContext('task');
-
-      navigateCrossDomain('https://valid.origin.com');
+    Object.values(FrameContexts).forEach(context => {
+      it(`navigation.navigateToTab should register the navigateToTab action when initialized with ${context} context`, () => {
```

Comment:
Same here. Actual API in pages file can check for the registration. Here we can mock a return value from pages.tabs.navifateToTab and make sure that we get true when it is completed successfully and an error message when there is an error.

Reply:
I see that you added code to check navigation.navigateToTab should successfully call pages.tabs.nagivateToTab. Don't we need to test .then which returns true and and .catch part for error?

Reply:
What I meant here was that we need to pass onComplete with error or with success case

Reply:
I agree that we might need a test that checks when we expect this sequence to fail/throw an error.